### PR TITLE
Help Center: clean-up article component

### PR DESCRIPTION
**Pull request by arcangelini**
Related to #92745

## Proposed Changes

This is a spinoff from the PR mentioned above. This refactors and simplifies the `/post` route that is responsible for displaying support guide articles in the Help Center

## Why are these changes being made?

1. The naming of the components was confusing and misleading
2. There seemed to be unnecessary abstraction
3. Duplicated imports

## Testing Instructions

1. Open Calypso and go to Help Center
2. Test articles and make sure there is no breakage
3. Be sure to try as free user as well to ensure Back button works.

### CODE REVIEW

File: packages/help-center/src/components/help-center-article.tsx [hunk 1]
```diff
@@ -68,25 +86,34 @@ export const HelpCenterEmbedResult: React.FC = () => {
...
+				<ArticleContent
+					content={ post?.content }
+					title={ post?.title }
+					link={ post?.link }
+					isLoading={ isLoading }
+					postId={ postId }
+					blogId={ blogId }
+					slug={ post?.slug }
+					articleUrl={ articleUrl }
+				/>{ ' ' }
```
Comment (agrullon95) [inline packages/help-center/src/components/help-center-article.tsx]: Do we need this `{ ' ' }`?

**Review by agrullon95 — APPROVED** [review]
LGTM, tested with paid and free users. I left a comment with question for the HelpCenterArticle component.

**Review by agrullon95 — APPROVED** [review]
LGTM, tested with paid and free users. I left a comment with question for the HelpCenterArticle component.

File: packages/help-center/src/components/help-center-article.tsx [hunk 2]
```diff
@@ -6,44 +6,62 @@ import { Icon, external } from '@wordpress/icons';
...
-	const link = params.get( 'link' );
+	const articleUrl = params.get( 'link' ) ?? undefined;
 	const query = params.get( 'query' );
+	const postKey = useSupportArticleAlternatePostKey( blogId, postId );
+	const post = usePostByKey( postKey ).data;
+	const isLoading = ! post?.content || ! postKey;
+
+	useEffect( () => {
+		//If a url includes an anchor, let's scroll this into view!
+		if ( articleUrl && articleUrl.indexOf( '#' ) !== -1 && post?.content ) {
```
Comment (escapemanuele) [inline packages/help-center/src/components/help-center-article.tsx]: Is it possible to use `articleUrl?.indexOf` here?

**Review by escapemanuele — APPROVED** [review]
Looking great! I left a minor comment, LGTM 👍

File: packages/help-center/src/components/help-center-article.tsx (hunk 2 above)
Comment (alshakero) [reply packages/help-center/src/components/help-center-article.tsx]: Maybe even `articuleUrl?.includes( '#' )`

**Pull request merged**